src/bloqade/submission/ir/task_specification.py

```diff
@@ -93,12 +218,54 @@ class Config:
     def __hash__(self):
         return hash((RabiFrequencyPhase, self.global_))
 
-    def discretize(self, task_capabilities: QuEraCapabilities):
+    def discretize(self, device_capabilities: QuEraCapabilities):
+        """
+        Discretizes the global field values based on the task capabilities.
+
+        Args:
+           device_capabilities (QuEraCapabilities): Capabilities of the task for discretization.
+
+        Returns:
+           RabiFrequencyPhase: Discretized Rabi frequency phase.
+
+        Example:
+            ```python
+            from decimal import Decimal
+            from bloqade.submission.ir.capabilities import QuEraCapabilities
+
+            # Example values for the global field
+            times = [Decimal('0.0'), Decimal('1.0'), Decimal('2.0')]
+            values = [Decimal('0.1'), Decimal('0.2'), Decimal('0.3')]
+
+            # Create a GlobalField
+            global_field = GlobalField(times=times, values=values)
+
+            # Create a RabiFrequencyPhase
+            rabi_frequency_phase = RabiFrequencyPhase(global_=global_field)
+
+            # Define example capabilities
+            class RydbergCapabilities:
+                class GlobalCapabilities:
+                    time_resolution = Decimal('0.5')
+                    phase_resolution = Decimal('0.1')
+
+                global_ = GlobalCapabilities()
+
+            class Capabilities:
+                rydberg = RydbergCapabilities()
+
+            device_capabilities = QuEraCapabilities(capabilities=Capabilities())
```

Comment:
This is not the intended useage of this function. I suggest using the default capabilities instead of creating your own for the examples. if you want to modify the capabilities, modify the capabilities object directly before passing it into the method. 

I would also make sure your examples run, this one is incomplete and doesn't run even if you have the correct imports.